timesketch/lib/ts_sigma.py

```diff
@@ -0,0 +1,177 @@
+    if not os.path.isfile(_CONFIG_FILE):
+        raise ValueError(
+            'Unable to open file: [{0:s}], it does not exist.'.format(
+                _CONFIG_FILE))
+
+    if not os.access(_CONFIG_FILE, os.R_OK):
+        raise ValueError(
+            'Unable to open file: [{0:s}], cannot open it for '
+            'read, please check permissions.'.format(_CONFIG_FILE))
+
+    with open(_CONFIG_FILE, 'r') as config_file:
+        sigma_config_file = config_file.read()
+
+    sigma_config = sigma_configuration.SigmaConfiguration(sigma_config_file)
+
+    return sigma_config
+
+def get_sigma_rules_path():
+    """Get the path for Sigma rules.
+
+        Args:
+            None
+
+        Returns:
+            A string
+    """
+
+    # TODO: Add functionality to have multiple paths for rule files.
+    _RULES_PATH = current_app.config.get('SIGMA_RULES_FOLDER')
+
+    if not _RULES_PATH:
+        raise ValueError(
+            'SIGMA_RULES_FOLDER not found in config file')
+
+    if not os.path.isdir(_RULES_PATH):
+        raise ValueError(
+            'Unable to open dir: [{0:s}], it does not exist.'.format(
+                _RULES_PATH))
+
+    if not os.access(_RULES_PATH, os.R_OK):
+        raise ValueError(
+            'Unable to open dir: [{0:s}], cannot open it for '
+            'read, please check permissions.'.format(_RULES_PATH))
+
+    return _RULES_PATH
+
+
+def get_sigma_rules(rule_folder):
+
+    return_array = []
+
+    for dirpath, dirnames, files in os.walk(rule_folder):
+        # TODO: lowercase the folder name in case a folder is named dePreCatEd
+        if 'deprecated' in dirnames:
+            dirnames.remove('deprecated')
+
+        for rule_filename in files:
+            if rule_filename.lower().endswith('yml'):
+                # if a sub dir is found, append it to be scanned for rules
+                if os.path.isdir(os.path.join(dirpath, rule_filename)):
+                    logger.info(
+                        'this is a directory, skipping: {0:s}'.format(
+                            rule_filename))
+                    continue
+
+                rule_file_path = os.path.join(dirpath, rule_filename)
+                parsed_rule = get_sigma_rule(rule_file_path)
+                return_array.append(parsed_rule)
+
+    return return_array
+
+
+def get_sigma_rule(filepath):
+    """ Returns a JSON represenation for a rule
+
+        Args:
+            filepath: path to the sigma rule to be parsed
+
+        Returns:
+            Json representation of the parsed rule
+        """
+
+
+    logger.info("get_sigma_rule with arg: {0:s}".format(filepath))
+    sigma_config = get_sigma_config_file()
+
+    sigma_backend = sigma_es.ElasticsearchQuerystringBackend(sigma_config, {})
+
+    if filepath.lower().endswith('yml'):
+        # if a sub dir is found, append it to be scanned for rules
+        if os.path.isdir(filepath):
+            logger.info(
+                'this is a directory, skipping: {0:s}'.format(
```

Comment:
same comment as above on the logger.info usefulness

Reply:
true, also improved the comment, as it was not accurate.